app/src/org/commcare/preferences/DeveloperPreferences.java

```diff
@@ -236,14 +196,16 @@ public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, Strin
                     DevSessionRestorer.clearSession();
                 }
                 setSessionEditText();
-
                 break;
-            case LOAD_FORM_PAYLOAD_AS:
-                if (!formLoadPayloadStatus().equals(FormRecord.STATUS_SAVED)) {
-                    // clear submission server so that 'unsent' forms that are loaded don't get sent to HQ
-                    CommCareApplication.instance().getCurrentApp().getAppPreferences().edit()
-                            .putString(CommCareServerPreferences.PREFS_SUBMISSION_URL_KEY, "")
-                            .apply();
+            case PREFS_CUSTOM_RESTORE_DOC_LOCATION:
```

Comment:
nice, great that we can move the handling of this here instead of `onActivityResult`